src/ucp/core/ucp_rkey.h

```diff
@@ -84,28 +96,39 @@ typedef struct {
  * The array itself contains only the MDs specified in md_map, without gaps.
  */
 typedef struct ucp_rkey {
-    /* cached values for the most recent endpoint configuration */
-    struct {
-        ucp_worker_cfg_index_t    ep_cfg_index; /* EP configuration relevant for the cache */
-        ucp_lane_index_t          rma_lane;     /* Lane to use for RMAs */
-        ucp_lane_index_t          amo_lane;     /* Lane to use for AMOs */
-        ssize_t                   max_put_short;/* Cached value of max_put_short */
-        uct_rkey_t                rma_rkey;     /* Key to use for RMAs */
-        uct_rkey_t                amo_rkey;     /* Key to use for AMOs */
-        ucp_amo_proto_t           *amo_proto;   /* Protocol for AMOs */
-        ucp_rma_proto_t           *rma_proto;   /* Protocol for RMAs */
-    } cache;
-    ucp_md_map_t                  md_map;       /* Which *remote* MDs have valid memory handles */
-    ucs_memory_type_t             mem_type;     /* Memory type of remote key memory */
-    uint8_t                       flags;        /* Rkey flags */
-    ucp_worker_cfg_index_t        cfg_index;    /* Rkey configuration index */
+    union {
+        /* cached values for the most recent endpoint configuration */
+        struct {
+            uint8_t                   flags;          /* Rkey flags */
+            ucp_worker_cfg_index_t    ep_cfg_index;   /* EP configuration relevant for the cache */
+            ucp_lane_index_t          rma_lane;       /* Lane to use for RMAs */
+            ucp_lane_index_t          amo_lane;       /* Lane to use for AMOs */
+            ucp_rkey_proto_index_t    amo_proto_index;/* Protocol for AMOs */
+            ucp_rkey_proto_index_t    rma_proto_index;/* Protocol for RMAs */
```

Comment:
[minor] pls add one more whitespace between `;` and comment (fix old bug, since code is touched)

Reply:
Fixed